src/reducers/preferences.js

```diff
@@ -0,0 +1,19 @@
+import { UPDATE_USER_PREFERENCES } from '../actions/preferences';
+
+const initialState = {
+  showDatasetPreviews: true,
+};
+
+const userPreferences = (state = initialState, action) => {
+  switch (action.type) {
+    case UPDATE_USER_PREFERENCES:
+      return {
+        ...state,
+        showDatasetPreviews: action.payload.preferences.showDatasetPreviews,
```

Comment:
```suggestion
        showDatasetPreviews: action.payload.showDatasetPreviews,
```

Reply:
The payload for userPreferences should be a preferences object. i.e., 
```
{
     showDatasetPreviews: false
}
```
It need not contain the preferences key inside like -
```
{
     preferences: {
           showDatasetPreviews: false
      }
}
```
I think there is some confusion around nesting. If we have payload as - `payload = showDatasetPreviews` (this means no nesting and cannot be extended in future)

`payload = {showDatasetPreviews: false}` , there is a nested prop and the payload becomes an object. Here, the object will have fields of User Preferences.